projects/pgai/pgai/vectorizer/embeddings.py

```diff
@@ -68,6 +89,92 @@ async def embed(
             embedding vectors or errors encountered during embedding.
         """
 
+    @abstractmethod
+    async def _inner_embed(self, documents: Documents) -> EmbeddingResponse:
+        """
+        TODO
+        :param documents:
+        :return:
+        """
+
+    @abstractmethod
+    def _max_chunks_per_batch(self) -> int:
+        """
+        The maximum number of chunks that can be embedded per API call
+        :return: int: the max chunk count
+        """
+
+    async def _outer_embed(self, documents: Documents) -> list[EmbeddingVector]:
```

Comment:
I'm not really happy with this split between _outer_embed and _inner_embed, because it is IMHO hard to follow.